core/math/math_defs.h

```diff
@@ -30,19 +30,29 @@
 
 #pragma once
 
+#include "core/typedefs.h"
+
+#include <limits>
+
+namespace Math {
+inline constexpr double SQRT2 = 1.4142135623730950488016887242;
+inline constexpr double SQRT3 = 1.7320508075688772935274463415059;
+inline constexpr double SQRT12 = 0.7071067811865475244008443621048490;
+inline constexpr double SQRT13 = 0.57735026918962576450914878050196;
+inline constexpr double LN2 = 0.6931471805599453094172321215;
+inline constexpr double TAU = 6.2831853071795864769252867666;
+inline constexpr double PI = 3.1415926535897932384626433833;
+inline constexpr double E = 2.7182818284590452353602874714;
+inline constexpr double INF = std::numeric_limits<double>::infinity();
+inline constexpr double NaN = std::numeric_limits<double>::quiet_NaN();
```

Comment:
We usually use uppercase for macros. I'm not sure off the top of my head if we have a standard for constants?

Reply:
We don't have a set standard for constants, but it's common to use uppercase if they are global scope and lowercase if they are within a function.

Reply:
Since these are all using `double`, won't this imply a conversion in code that uses floats?
I tried in Compiler Explorer and it seems to make a difference: https://godbolt.org/z/hj14W3Kdc

Reply:
It would imply runtime conversion, but that was already the case with the previous `#define` method. This cost is circumvented with the appropriate cast, though warnings for this kind of conversion are currently disabled across the repo